chia/full_node/full_node.py

```diff
@@ -595,7 +595,9 @@ async def short_sync_batch(self, peer: WSChiaConnection, start_height: uint32, t
                 self.sync_store.batch_syncing.remove(peer.peer_node_id)
                 self.log.error(f"Error short batch syncing, could not fetch block at height {start_height}")
                 return False
-            if not self.blockchain.contains_block(first.block.prev_header_hash):
+            hash = self.blockchain.height_to_hash(first.block.height - 1)
+            assert hash is not None
```

Comment:
You want to just assert here or raise something specific string?

Reply:
@almogdepaz if we're actually backtracking from a new fork, I believe this can *legitimately* return `None`, because `height - 1` may be greater than our current peak. I think Earle's suggestion would fix that

Reply:
this is short_sync_batch we do not backtrack we start from a fork point that we assume is in our chain, if it is not we return false and do a long_sync since its a reorg

Reply:
ok. the `start_height` that's passed in is guaranteed to be at most one-past the current peak. Technically, the peer could respond with an invalid block whose height is differemt. But I suppose in that case, it's fine to assert and (probably) just tear down the whole connection.

Reply:
we need to either assert or throw here so we disconnect and ban the peer

Reply:
I think `assert` communicates that there's a bug if it fires, whereas checking and throwing communicates that it's a failure that can happen if the peer sends an invalid message. So, not asserting would be slightly less confusing I think.

Reply:
I think I'd rather raise something specific `raise ValueError("well that's not right")` - or some such. but it's not that important